test/e2e/storage/csi_volumes.go

```diff
@@ -83,13 +86,18 @@ var _ = utils.SIGDescribe("CSI Volumes", func() {
 			WaitForCompletion: true,
 		}
 		csiDriverRegistrarClusterRole(config)
+		createCSICRDs(crdclient)
+	})
+
+	AfterEach(func() {
+		deleteCSICRDs(crdclient)
 	})
 
 	for driverName, initCSIDriver := range csiTestDrivers {
 		curDriverName := driverName
 		curInitCSIDriver := initCSIDriver
 
-		Context(fmt.Sprintf("CSI plugin test using CSI driver: %s", curDriverName), func() {
+		Context(fmt.Sprintf("CSI plugin test using CSI driver: %s [Serial]", curDriverName), func() {
```

Comment:
why do these tests need to be serial? AFAIK they weren't changed?

CRD creation can happen in the other context "CSI attach test using HostPath driver [Serial][Feature:CSISkipAttach]" Since that's all that would need it.

Reply:
Adding on to that, is there any way to generalize the tests you wrote below in ""CSI attach test using HostPath driver [Serial][Feature:CSISkipAttach]" to work for all drivers in this suite? It's kind of awkward to have this very general framework up here and then a random feature test with just hostpath below it. If it has to be only hostpath, then I would recommend putting it in a different file `csi_hostpath_attach.go` for example

Reply:
cc @jsafrane who wrote the test